agcaballero/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -47,31 +56,56 @@ public String toString() {
     }
   }
 
-  // holds all comments that have been submitted in session so far
-  private final ArrayList<Comment> comments = new ArrayList<Comment>();
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Retrieve from Datastore entities of type "Comment"
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
```

Comment:
I'm curious to know if the query or property fetch is case sensitive. I see capital "Comment" here but lowercase "comment" below.

Reply:
I think it must be. The "Comment" and "comment" actually refer to different things. "Comment" refers to the kind of entity, and "comment" is a property of this kind that actually holds the text. I think I'll change the variable name to "text" to make this more clear. Thank you!

Reply:
Done